### ISSUE TITLE
[Go] Implement new Concept Exercise: time

### ISSUE THREAD

**Issue by oanaOM** [exercism/v3#1925]
This issue describes how to implement the time concept exercise for the Go track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to use time in Go.

## Learning objectives

- Know how to use time.Time in Go.
- Know how to get the current date and time.
- Know how to perform compare time.
- Know how to parse date and time from a string.

## Out of scope

- timezones

## Concepts

- `time`

## Prerequisites

- `basics`
- `numbers`
- `strings`
- `functions`

### Hints

- [Time package](https://golang.org/pkg/time/)
- [Parsing dates from a string and formatting](https://golangcode.com/parsing-dates/)

## Analyzer

This exercise does not require any specific logic to be added to the analyzer.

## Implementing

To implement this exercise, please follow these [instructions](https://github.com/exercism/v3/blob/master/languages/go/reference/implementing-a-concept-exercise.md) and the below file structure.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by jamessouth** [comment]
I can take this one

**Comment by oanaOM** [comment]
awesome! All yours @jamessouth

**Comment by jamessouth** [comment]
I want to note that this is apparently the first Go topic so far that is not a basic built-in concept like conditionals. It is not the first to use methods imported from outside packages (for instance, the `strings` Concept imports from the `strings` package), so I think for now I will assume the student, having reached the `time` (or other precursor) Concept, can:
- read the docs to find what they need
- understand how to import, or understand that their IDE/in-browser coding environment is doing it for them
- differentiate between methods called on instances of a type and methods called on the package (e.g. `myTimeVar.After()` vs `time.Now()`)

**Comment by jamessouth** [comment]
I'm adapting this from the csharp datetime CE, which I think will cover the objectives listed above, but not the Go `time.Duration` type. Should it? Or leave for a later `time` CE, which might also cover timers/tickers, stop, etc., and time zones/location, which is o/o scope for this one?

**Comment by tehsphinx** [comment]
@jamessouth: thx a lot! I merged the new exercise.

I think we should have a second exercise that introduces `time.Duration`. @oanaOM: Do you want to create another issue for that?

Some things I think it should cover:
- relation between time.Time and time.Duration
- time measuring (e.g. execution time of a function -- no `defer`!) (time.Since)
- diff between 2 times
- time units like 4 * time.Second
- time.Add
- time.Sleep (?)

**Comment by oanaOM** [comment]
@tehsphinx sure! I've open #2072 
I'll close this one since it's done.
